components/camera/camera_test.go

```diff
@@ -223,6 +223,27 @@ func TestReconfigurableCamera(t *testing.T) {
 	test.That(t, stream.Close(context.Background()), test.ShouldBeNil)
 }
 
+func TestSourceFromCamera(t *testing.T) {
+	videoSrc := &simpleSource{"rimage/board1_small"}
+	actualCam, err := camera.NewFromReader(context.Background(), videoSrc, nil, camera.UnspecifiedStream)
+	test.That(t, err, test.ShouldBeNil)
+
+	reconfCam, err := camera.WrapWithReconfigurable(actualCam, resource.Name{})
+	test.That(t, reconfCam, test.ShouldNotBeNil)
+	test.That(t, err, test.ShouldBeNil)
+
+	source, err := camera.SourceFromCamera(reconfCam.(camera.Camera))
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, source, test.ShouldNotBeNil)
+
+	wgCam := camera.CameraWaitGroup{Cam: reconfCam.(camera.Camera)}
+	test.That(t, wgCam.Cam, test.ShouldNotBeNil)
+
+	source2, err := camera.SourceFromCamera(wgCam.Cam)
```

Comment:
You shouldn't need to pull out the `Cam` here. If you do you aren't testing that this function actually works with a `CameraWaitGroup` but are retesting that it works with a reconfigurable camera since `Cam = reconfCam`